### DISCUSSION TITLE
Conectar a placa  MKS DLC32 v2.1 a universal g-code sender

### DISCUSSION THREAD

**Discussion by Gabriel28Anselmo** [winder/Universal-G-Code-Sender#2381] [Help]
@breiler boa tarde como esta?
Breiler, tenho a placa  MKS DLC32 v2.1 e hoje utilizo o software LigthBurn para ler e executar os meus arquivos G-code.
porem, estou sentindo necessidade de migrar de software do LightBurn para o Universal G-code Sender Platform.
a minha placa  MKS DLC32 v2.1 possibilita conexão via wifi e porta com. porem nao estou conseguindo conectar o software UGS à placa.
configurei instruções que você mencionou para o outro colaborador do grupo 
https://github.com/winder/Universal-G-Code-Sender/wiki/Connecting-the-Controller#smoothie

fiz e porem não consegui conectar a placa 

utilizo o grbl como firmware 
e tenho a opção de conexão via wifi e porta usb 
consegue me dar uma ajuda?
sou o Gabriel Anselmo desenvolvedor iniciante aqui do Brasil 

desde já agradeço

**Comment by Gabriel28Anselmo** [comment]
![image](https://github.com/winder/Universal-G-Code-Sender/assets/153220619/8aafbca3-00e0-49ab-9224-5b0b9afcaf98)

**Comment by breiler** [comment]
Start by connecting to it from USB, what is the console output when you try to connect to it?

**Reply by Gabriel28Anselmo** [reply]
com12

**Reply by breiler** [reply]
Are you saying that this window only writes com12?

![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/f5e2794a-8dae-4226-8630-6e649b2cba32)

**Comment by Gabriel28Anselmo** [comment]
![image](https://github.com/winder/Universal-G-Code-Sender/assets/153220619/b7700f9c-0c0b-4c40-be10-6a4a49c05aba)

essas são as opções que tenho na configuração

**Reply by breiler** [reply]
There are so many errors with your settings...

Start with connecting to it using USB serial to make sure that UGS can talk with the board.
* Select the connection driver `JSerialComm`
* Select the firmware `GRBL` (I assume that it is installed on your board and not Smoothieware)
* Select the port in the drop down (you mentioned COM15 earlier)
* Set the baud to `115200`
* Connect to the board
* Copy and paste the output from the console here

**Reply by Gabriel28Anselmo** [reply]
*** Connecting to jserialcomm://COM12:115200
*** Fetching device status
>>> ?
buffsize:243
<Idle|MPos:0.000,0.000,0.000|Bf:63,243|FS:0,0|Pn:PXYZ>
ok
*** Fetching device status (2 of 3)...
>>> ?
buffsize:243
<Idle|MPos:0.000,0.000,0.000|Bf:63,243|FS:0,0|Pn:PXYZ>
ok
*** Fetching device status (3 of 3)...
>>> ?
buffsize:243
<Idle|MPos:0.000,0.000,0.000|Bf:63,243|FS:0,0|Pn:PXYZ>
ok
**** Connection closed ****

**Reply by Gabriel28Anselmo** [reply]
![image](https://github.com/winder/Universal-G-Code-Sender/assets/153220619/6a5fdad7-f94e-4220-9eb5-7a9cb1bfb60c)

**Reply by breiler** [reply]
All the settings looks correct and you are getting some correct responses from the controller
except for the "buffsize:243" which is not a part of the GRBL protocol and will prevent you from using UGS.

I looked through the code of the MKS firmware and couldn't find any reference of "buffsize": 
https://github.com/makerbase-mks/MKS-DLC32-FIRMWARE/tree/main/Firmware

I can't help you with this board.
